frontend-react/src/components/Admin/AdminReceiverDashboard.tsx

```diff
@@ -750,13 +984,15 @@ export function AdminReceiverDashboard() {
             </form>
             <MainRender
                 data={data}
-                filterRowStatus={filterRowSuccessState}
                 datesRange={[
                     new Date(startDate),
                     endDate
                         ? new Date(endOfDayIso(new Date(endDate)))
                         : new Date(endOfDayIso(new Date())),
                 ]}
+                filterRowStatus={filterRowSuccessState}
```

Comment:
Yeah, pushing the fetch down into sub-components is a great idea. We should do it in more places. I hate how all the page UI vanishes when there's a refresh.

I put it at the top because it makes testing components a bit easier because of rest-hook mocks.

I'll see how big of a refactor it is. If need be, I can always do another quick refactor to fetch deeper down. Might be nice to break up these big changes anyways.

Reply:
This was easier and cleaner than expected. Thanks for the suggestion. Unit tests now mock the data at the rest-hook level.